UOP1_Project/Assets/Scripts/Characters/StateMachine/Actions/FaceProtagonistSO.cs

```diff
@@ -0,0 +1,37 @@
+﻿using UnityEngine;
+using UOP1.StateMachine;
+using UOP1.StateMachine.ScriptableObjects;
+
+[CreateAssetMenu(fileName = "FaceProtagonist", menuName = "State Machines/Actions/Face Protagonist")]
+public class FaceProtagonistSO : StateActionSO
+{
+	protected override StateAction CreateAction() => new FaceProtagonist();
+}
+
+public class FaceProtagonist : StateAction
+{
+	Transform _protagonist;
+	Transform _actor;
+
+	public override void Awake(StateMachine stateMachine)
+	{
+		_actor = stateMachine.transform;
+	}
+		
+	public override void OnUpdate()
+	{
+		if (_protagonist != null)
+		{
+			_actor.LookAt(_protagonist);
+		}
+	}
+	
+	public override void OnStateEnter()
+	{
+		_protagonist = GameObject.FindObjectOfType<Protagonist>().transform;
```

Comment:
You should cache this on `Awake`, or, actually, it should be in a field in `FaceProtagonistSO` so it can be set from the inspector. Also, I'd rename the action to `FaceTarget` since this could be useful for other things as well.

Reply:
Actually, I initially put this in Awake, but it does not work because here the protagonist is instantiated by the spawning system after Awake is triggered. 

I was thinking also about having a target more generic not Protagonist specific, but how do you specify a target that has not been instantiated yet in the SO?

Reply:
> Actually, I initially put this in Awake, but it does not work because here the protagonist is instantiated by the spawning system after Awake is triggered.
Hmm, I've actually had this at the back of my mind for a while... The **SM** is calling `Awake` on the initial state _during_ the **SM** `Awake`. I think it should be called on `Start` to solve this.

Reply:
@treivize that's a good question.

Actually, one of the last things I implemented might be an answer to this. [I mentioned it here](https://forum.unity.com/threads/decouple-cameramanager-and-spawnsystem-card-194.1016932/#post-6592348).

You could use one of those to connect this ActionSO to the player's transform, even if you don't know yet if it's been instantiated or not. You could just implement a check to see if it's null. If it's not, it will be the player's transform.

The check is also not super necessary, since it will only happen if the player has entered the TriggerZone, so the player exists at that point.